File: forge/src/main/java/org/valkyrienskies/eureka/forge/DeferredRegisterBackend.java
package org.valkyrienskies.eureka.forge;

import net.minecraft.core.Registry;
import net.minecraft.resources.ResourceKey;
import org.jetbrains.annotations.NotNull;
import org.valkyrienskies.eureka.registry.DeferredRegister;

public class DeferredRegisterBackend implements DeferredRegister.DeferredRegisterBackend {

    @NotNull
    @Override
    public <T> DeferredRegister<T> makeDeferredRegister(@NotNull String id, @NotNull ResourceKey<Registry<T>> registry) {
        return new DeferredRegisterImpl(id, registry);
    }
}

File: common/src/main/kotlin/org/valkyrienskies/eureka/services/DeferredRegisterBackend.kt
package org.valkyrienskies.eureka.services

import net.minecraft.core.Registry
import net.minecraft.resources.ResourceKey
import org.valkyrienskies.eureka.registry.DeferredRegister

interface DeferredRegisterBackend {
    fun <T> makeDeferredRegister(id: String, registry: ResourceKey<Registry<T>>): DeferredRegister<T>
}